site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,694 @@
+bot.command("start", async (ctx) => {
+  await ctx.reply(ctx.t("greeting"));
+});
 
-bot.start();
+bot.command("language", async (ctx) => {
+  if (ctx.match === "") {
+    return await ctx.reply(ctx.t("language.specify-a-locale"));
+  }
+
+  // `i18n.locales` contains all the locales that have been registered
+  if (!i18n.locales.includes(ctx.match)) {
+    return await ctx.reply(ctx.t("language.invalid-locale"));
+  }
+
+  // `ctx.i18n.getLocale` returns the locale currently using.
+  if (await ctx.i18n.getLocale() === ctx.match) {
+    return await ctx.reply(ctx.t("language.already-set"));
+  }
+
+  await ctx.i18n.setLocale(ctx.match);
+  await ctx.reply(ctx.t("language.language-set"));
+});
 ```
 
-In order to have full TypeScript support you also need to adapt your [Context Flavors](../guide/context.md#context-flavors).
-Check the link for more details on how they work.
-Here is a short example:
+</CodeGroupItem>
+ <CodeGroupItem title="JavaScript">
+
+```js
+const { Bot, session } = require("grammy");
+const { I18n } = require("@grammyjs/i18n");
+
+const i18n = new I18n({
+  defaultLocale: "en",
+  useSession: true, // whether get/set in session
+
+  // Load locales from the `locales` directory.
+  directory: "locales",
+});
+
+const bot = new Bot(""); // <-- put your bot token here
+
+// Remember to register `session` middleware before
+// registering middleware of the i18n instance.
+bot.use(session({
+  initial: () => {
+    return {};
+  },
+}));
+
+// Register i18n middleware
+bot.use(i18n.middleware());
+
+bot.command("start", async (ctx) => {
+  await ctx.reply(ctx.t("greeting"));
+});
+
+bot.command("language", async (ctx) => {
+  if (ctx.match === "") {
+    return await ctx.reply(ctx.t("language.specify-a-locale"));
+  }
+
+  // `i18n.locales` contains all the locales that have been registered
+  if (!i18n.locales.includes(ctx.match)) {
+    return await ctx.reply(ctx.t("language.invalid-locale"));
+  }
+
+  // `ctx.i18n.getLocale` returns the locale currently using.
+  if (await ctx.i18n.getLocale() === ctx.match) {
+    return await ctx.reply(ctx.t("language.already-set"));
+  }
+
+  await ctx.i18n.setLocale(ctx.match);
+  await ctx.reply(ctx.t("language.language-set"));
+});
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="Deno">
 
 ```ts
-import { Context as BaseContext, SessionFlavor } from "grammy";
-import { I18nContextFlavor } from "@grammyjs/i18n";
+import {
+  Bot,
+  Context,
+  session,
+  SessionFlavor,
+} from "https://deno.land/x/grammy/mod.ts";
+import {
+  I18n,
+  I18nContextFlavor,
+} from "https://deno.land/x/grammy_i18n/mod.ts";
 
-interface Session {
-  apples?: number;
+interface SessionData {
+  __language_code?: string;
 }
 
-export type MyContext =
-  & BaseContext
-  & SessionFlavor<Session>
+type MyContext =
+  & Context
+  & SessionFlavor<SessionData>
   & I18nContextFlavor;
+
+const i18n = new I18n({
+  defaultLocale: "en",
+  useSession: true, // whether get/set in session
```

Comment:
```suggestion
  useSession: true, // whether to store user language in session
```

Reply:
thats a good example of wording it less technical but not tutorial like :)

Reply:
What problem do you even have with tutorials? We have loads of them on this website, Getting Started is probably the most famous one.

Reply:
So should this be more like tutorial and the api doc the core doc?

Reply:
I don't know what difference there is between docs and tutorials from your point of view.